file-formats/dsfi/type/zif_aff_dsfi_v1.intf.abap

```diff
@@ -2,18 +2,36 @@ INTERFACE zif_aff_dsfi_v1
   PUBLIC.
 
 
-  "! <p class="shorttext">Engine</p>
-  "! Engine
-  "! $values {@link zif_aff_dsfi_v1.data:co_engine}
-  TYPES: ty_engine TYPE zif_aff_types_v1=>ty_object_name_30.
+  TYPES:
+    "! <p class="shorttext">Engine</p>
+    "! Engine
+    "! $values {@link zif_aff_dsfi_v1.data:co_engine}
+    "! $default {@link zif_aff_dsfi_v1.data:co_engine.analytical_engine}
+    ty_engine TYPE zif_aff_types_v1=>ty_object_name_30.
 
   CONSTANTS:
     BEGIN OF co_engine,
       "! <p class="shorttext">Analytical Engine</p>
       "! Analytical engine
       analytical_engine TYPE ty_engine VALUE 'ANA',
+      "! <p class="shorttext">SQL Engine</p>
+      "! SQL engine
+      sql_engine TYPE ty_engine VALUE 'SQL',
     END OF co_engine.
 
+  TYPES:
+    BEGIN OF ty_sql_properties,
+      "! <p class="shorttext">AMDP Reference</p>
+      "! The name of the AMDP implementation reference
+      "! $required
+      amdp_name          TYPE c LENGTH 100,
+      "! <p class="shorttext">Auto Exposure</p>
+      "! Flag whether or not the SQL implementation of the
+      "! CDS scalar function will be automatically exposed in all SQL services
+      "! $required
+      auto_exposure_flag TYPE abap_bool,
```

Comment:
Can you explain what the exposure field means. I discussed this with @wurzka and from UX perspective we struggle above two aspects:

1. We think the addition `_flag` is not needed here. This is a quite technical term. We suggest to remove it.
2. What does **automatically** mean? Is there also a manual way to expose the SQL implementation in a SQL service?

Reply:
1. I agree, but analogously to the `amdp_name` discussion, we'd like to keep the standard code consistent across the board and the `_flag` addition is present throughout the system for this field.
2. SQL implementations usually **must** be exposed manually in a SQL service. The auto-exposure is a mechanism to prevent repetition for commonly used functions. They would otherwise need to be explicitly declared in many services, and the services themselves would become bloated with repeated function exposures.

Reply:
As discussed, we've agreed to match the field names to their descriptions.